docs/assignment3.md

```diff
@@ -1 +1,306 @@
+- HashMap<Integer, ArrayList<Team>> getWinners(): calls *playRound* method and returns *winners*  
+-  *int getId()*: returns *leagueId*  
+  
+Associations  
+  
+A single **League** object can exist at one time within a **Game**. A league can start with at most 16 teams as per requirement **FR6**. Within the *playRound* method, **Match** objects are created from the playing teams to generate a winner.  
+  
+#### **Match**  
+  
+A match simulates a single soccer game between two teams.  
+  
+Attributes  
+  
+* *team1* (Type: Team): first team in the match  
+* *team2* (Type: Team): second team in the match  
+  
+Operations  
+  
+- *Team getWinner()*: returns the winner of a match as described in **FR8** and **FR9**  
+  
+Assiciations  
+  
+A match is formed of exactly 2 **Team** objects.  A match can belong to only one league at most, and logically, a league can not spawn more than 8 **Match** objects at one time.
+  
+### *Views* package  
+  
+Each class in this package is an extention of the **GamePane** abstract class, described below. Each uses the *javafx* library and represents a different window in the GUI. All interactions with the *models* package are made via the **Game** class methods.  
+  
+#### **GamePane**  
+  
+Abstract methods  
+  
+- *void connectComponents()*: Connects all components into the root of a *javafx.scene.Scene*  
+- *void styleComponents()*: Styles all components as required  
+- *void setCallbacks()*: Sets callbacks for all interactive components  
+  
+### *Controllers* package  
+  
+Only consists of class **SceneManager** used for managing the *javafx* Scenes.  
+  
+Operations  
+  
+- *void setStage(final Stage stage)*: Sets the primary stage  
+- *void showScene(final Scene scene)*: Replaces the currently active Scene with another one  
+- *void showPane(final Class<? extends GamePane> pane)*: Replaces the current  **GamePane** with another  
+- *showPane(final Class<? extends GamePane> pane)*: Returns coresponding **GamePane** 
+- *SceneManager getInstance()*: returns the instance of the class
+
+
+## Object diagrams								
+Author(s): `Torstein Thomassen`
+
+This object diagram shows an example of objects with different states in our program.
+All the different attributes and variables has be fulled in with example data.
+
+This structure resembles a tree structure with branches and leafs. The tree structure is commonly used in many software design. It's an efficient and a good way to structure the relation between objects. 
+
+This creates a hierarchy where a parent can only directly affect the children. E.g the **Match** can't affect the **Player** 'directly', only the **Team** objects.       
+
+`Figure representing the UML object diagram`
+  
+![](imgs/ass3%20objDiagram.png)
```

Comment:
The Object Diagram's attributes are incorrectly modelled  -> From the slides:
![image](https://user-images.githubusercontent.com/59608110/78667282-f6605980-78d8-11ea-8eff-dd1b1110ca87.png)
The '=' is missing, even when example values are assigned, eg. Name and Score for the Player objects.

Reply:
Indeed, take this as TA feedback